txn/flusher.go

```diff
@@ -212,6 +212,8 @@ var txnFields = bson.D{{"txn-queue", 1}, {"txn-revno", 1}, {"txn-remove", 1}, {"
 
 var errPreReqs = fmt.Errorf("transaction has pre-requisites and force is false")
 
+const maxTxnQueueLength = 1000
```

Comment:
This seems like a sensible default but it might be good to add support for overriding it on a per session basis in case it's required for some application.